Домашняя работ №-9

NickFilimonov opened the pull request.

NickFilimonov committed:
pr. Playlist Maker commit 1

NickFilimonov committed:
pr. Playlist Maker commit 2

NickFilimonov committed:
pr. Playlist Maker commit 3

NickFilimonov committed:
pr. Playlist Maker commit 4

NickFilimonov committed:
pr. Playlist Maker commit 5

NickFilimonov committed:
pr. Playlist Maker commit 6

NickFilimonov committed:
pr. Playlist Maker commit 7

NickFilimonov committed:
pr. Playlist Maker commit 8

NickFilimonov committed:
pr. Playlist Maker commit 9

NickFilimonov committed:
pr. Playlist Maker commit 10

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,71 @@
+        <item name="android:layout_width">300dp</item>
+        <item name="android:layout_height">19dp</item>
+        <item name="android:layout_marginLeft">@dimen/margin_horizontal</item>
+        <item name="android:layout_marginTop">21dp</item>
```
Лучше внешние отступы, высоту и ширину, веса не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,71 @@
+
+    <style name="search_activity_button" parent="Widget.MaterialComponents.Button.UnelevatedButton">
+        <item name="android:layout_width">0dp</item>
+        <item name="android:layout_height">55.5dp</item>
```
Не стоит использовать дробные dp

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,71 @@
+        <item name="android:layout_marginStart">16dp</item>
+    </style>
+
+    <style name="MySwitch">
```
Стоит делать названия единообразными. Сейчас некоторые стили написаны в snake_case а некоторые - в PascalCase

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,71 @@
+        <item name="android:layout_height">0dp</item>
+        <item name="android:layout_weight">1</item>
+        <item name="android:insetTop">0dp</item> //несмотря на то, что  стиль кнопок унаследован от UnelevatedButton,
+        <item name="android:insetBottom">0dp</item> // без этих атрибутов у кнопки появляются отсутпы от границ View сверху и снизу
```
в xml комментарии делаются по-другому:` <!--вот так-->`

faritowich commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,16 @@
+    <string name="message_for_developers">Спасибо разработчикам и разработчицам за крутое приложение!</string>
+    <string name="subject_for_developers">Сообщение разработчикам и разработчицам приложения Playlist Maker</string>
+    <string name="practicum_offer">https://yandex.ru/legal/practicum_offer/</string>
+    <string name="email_support">nfilimonov1@gmail.com</string>
```
Чтобы не ругалось на отсутствие перевода, стоит или удалить папку с strings en, или перевести на англ, или добавить translatable="false"

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,86 @@
+        setContentView(R.layout.activity_search)
+
+        val clearButton = findViewById<ImageView>(R.id.clear_search_button)
+        val editText = findViewById<EditText>(R.id.edit_text_search)
```
Сейчас для вью R.id.edit_text_search переменная объявляется дважды в разных местах. Этого можно избежать, если объявлять глобальную переменную в теле класса с lateinit (будет private lateinit var viewName: EditText) или как нуллабл переменную (будет private var viewName: EditText? = null, (используется реже)), а в onCreate уже задавать значение findViewById<EditText>(R.id.VIEW_ID). Все остальные переменные, отвечающие за вьюхи, можно объявить так же. Тогда переменную можно инициализировать один раз в onCreate и иметь к ней доступ из любой функции класса.

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,86 @@
+}
+    // Метод делает видимой или не видимой кнопку clearButton
+    // в зависимости от того есть или нет текст в поле editText
+    fun clearButtonVisibility(s: CharSequence?): Int {
```
private

NickFilimonov committed:
pr. Playlist Maker commit 11
- Исправил верстку для activity_settings и activity_search (добавил кнопку "назад", убрал кнопки переходов между экранами внизу)
- Вынес внешние отступы и размеры из стилей в активити
- Убрал дробные размеры
- Сделал название стилей единообразными snake_case
- Удалин файл en-strings
- Объявил переменную editText глобально в теле класса
